### PR TITLE
Adding Dashboard to a component

### PR DESCRIPTION
**Pull request by chmst**
One of the core concepts is the Dashboard for components. This PR describes how to add a dashboard to an own component. 

ToDo: Add screenshots.

### CODE REVIEW

File: docs/using-core-functions/Dashboard.md [hunk 1]
```diff
@@ -0,0 +1,145 @@
+It is common usage to insert a link into the component menu for direct access to your own component.
+This is done in the manifest file: 
+
+```HTMLx title="Menu item Component in example.xml "
```
Comment (HLeithner) [inline docs/using-core-functions/Dashboard.md:4]:
```suggestion
```xml title="Menu item Component in example.xml "
```

**Comment by HLeithner** [comment]
can your replace "HTMLx" with "xml" thanks

File: docs/using-core-functions/Dashboard.md [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+
+A parameter < dashboard > expands the menu entry with link to a dashboard for your component. You can give this dashboard any name you want, but <strong> Note:</strong> use lowercase and only "-", never underscore for the dashboard name. 
+my-example or example are correct, my_example, Com-MY_EXAMPLE are wrong.
+
+The param < dashboard > 
+- will make a dashboard icon appear next to the administrator menu item for the component
+- The dashboard icon will click through to display modules assigned to the cpanel-example administrator module position
+- The title and icon defined in the XML file will be used as the header and icon at the top of the component's dashboard page.
+
+```HTMLx title="Dashboard Link "
```
Comment (HLeithner) [inline docs/using-core-functions/Dashboard.md:22]:
```suggestion
```xml title="Dashboard Link "
```

File: docs/using-core-functions/Dashboard.md [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+	[..]
+</administration>
+```
+Joomla provides now a dashboard for your component. You can add modules here using the <strong> position: cpanel-example </strong>. 
+
+## Dashboard title and icon 
+
+Give your dashboard a name and an icon. Add this to your manifest file:
+
+```HTMLx title="Dashpoard title and icon"
```
Comment (HLeithner) [inline docs/using-core-functions/Dashboard.md:40]:
```suggestion
```xml title="Dashpoard title and icon"
```

File: docs/using-core-functions/Dashboard.md [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+    <dashboards>
+		<dashboard title="COM_EXAMPLE" icon="icon-calendar">COM_EXAMPLE_DASHBOARD_NAME</dashboard>
+	</dashboards>
+```
+
+## Submenu
+If you want to address different views of your component, expand the menuy by a submenu.  
+
+``` HTMLx title="Submenu items"
```
Comment (HLeithner) [inline docs/using-core-functions/Dashboard.md:49]:
```suggestion
```xml title="Submenu items"
```

File: docs/using-core-functions/Dashboard.md [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+- add the presets folder to your .xml file
+- expand your install script
+
+Presets are already used in the core, see examples the component com_menu.
+
+## The menu preset
+
+In a folder 'presets' write a preset file, name it example.php. 
+
+``` HTMLx title="presets/example.xml"
```
Comment (HLeithner) [inline docs/using-core-functions/Dashboard.md:80]:
```suggestion
```xml title="presets/example.xml"
```

File: docs/using-core-functions/Dashboard.md [hunk 6]
```diff
@@ -0,0 +1,145 @@
...
+			link="index.php?option=com_categories&amp;view=categories&amp;extension=com_example"
+		/>
+	</menuitem>
+</menu>
+```
+
+
+## Manifest file
+
+``` HTMLx title="New folder in example.xml
```
Comment (HLeithner) [inline docs/using-core-functions/Dashboard.md:113]:
```suggestion
```xml title="New folder in example.xml
```

File: docs/using-core-functions/Dashboard.md [hunk 7]
```diff
@@ -0,0 +1,145 @@
...
+	</administration>
+```
+
+## install script
+
+We suppose here that you have a install script in your component. If not, please read the doc for install scripts.
+
+You add your preset in the dashboard of your component with a single line of code during installation: 
+
+``` HTMLx title="installation script"
```
Comment (HLeithner) [inline docs/using-core-functions/Dashboard.md:139]:
```suggestion
```xml title="installation script"
```

File: docs/using-core-functions/Quicktasklink.md [hunk 8]
```diff
@@ -0,0 +1,64 @@
+Quicktask Link and Icon
+=======================
+
+In general, a component has has one or more links to its views.They are defined in the manifest file and is added int othe menu during installation.
+
+```HTMLx title="A link to your component "
```
Comment (HLeithner) [inline docs/using-core-functions/Quicktasklink.md:6]:
```suggestion
```xml title="A link to your component "
```

File: docs/using-core-functions/Quicktasklink.md [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+```
+Sometimes it is useful to add so called <strong>quicktask link</strong> which enables an action calling fromthe menuy without first visiting the Overview.
+You can see this for Joomla core components, for example articles in com_content, here the quicktask is the plus icon and lets you add a new article in a single click.
+Your menu link is added in the manfest file during installation.
+
+## Qicktask link
+
+A quicktask link and icon are added as params to a menu item.
+
+``` HTMLx title="Qicktask Link and icon"
```
Comment (HLeithner) [inline docs/using-core-functions/Quicktasklink.md:23]:
```suggestion
```xml title="Qicktask Link and icon"
```

File: docs/using-core-functions/Quicktasklink.md [hunk 10]
```diff
@@ -0,0 +1,64 @@
...
+		</params>
+	</menu>
+
+```
+
+## Example 
+
+This example shows a complete menu with dashboard, submenu and a quicktask.
+
+```
```
Comment (HLeithner) [inline docs/using-core-functions/Quicktasklink.md:38]:
```suggestion
```xml
```

**Comment by HLeithner** [comment]
@chmst any chance to complete this pr in the near future? No problem if not just wanted to know

**Comment by chmst** [comment]
Sorry for delay, will do asap

**Comment by chmst** [comment]
@HLeithner thanks for suppot. I think it is ready now.

**Pull request merged**